src/productcatalogservice/server.go

```diff
@@ -164,6 +169,27 @@ func initStackDriverTracing() {
 	log.Warn("could not initialize stackdriver exporter after retrying, giving up")
 }
 
+// Initialize OTel trace provider that exports to Cloud Trace
+func initTraceProvider() {
+	// Initialize exporter OTel Trace -> Google Cloud Trace
+	projectID := os.Getenv("GOOGLE_CLOUD_PROJECT")
+	exporter, err := texporter.NewExporter(texporter.WithProjectID(projectID))
+	if err != nil {
+		log.Fatalf("failed to initialize exporter: %v", err)
+	}
+
+	// Create trace provider with the exporter.
+	tp, err := sdktrace.NewProvider(sdktrace.WithConfig(
+		sdktrace.Config{DefaultSampler: sdktrace.AlwaysSample()}),
+		sdktrace.WithSyncer(exporter),
+		// TODO: replace with predefined constant for GKE or autodetection when available
+		sdktrace.WithResource(resource.New(standard.ServiceNameKey.String("GKE"))))
+	if err != nil {
+		log.Fatal("failed to initialize trace provider: %v", err)
```

Comment:
I don't think we should fail the app if tracing doesn't work, but we should log it as critical error.

Reply:
yep, got it :)